File: icommerce/src/main/java/com/icommerce/modelo/Fichaje.java
package com.icommerce.modelo;


import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

import javax.persistence.Table;
import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.Data;

/**
 *
 * @author Miguel
 */
@Data
@Entity
@Table(name = "fichaje")
public class Fichaje {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @JsonFormat(pattern="yyyy-MM-dd HH:mm:ss")
    @Column(name = "hora_entrada")
    private Date horaEntrada;
    
    @Column(name = "activo")
    private Boolean activo;
    
    @JsonFormat(pattern="yyyy-MM-dd HH:mm:ss")
    @Column(name = "hora_salida")
    private Date horaSalida;
    
    
    @ManyToOne
    @JoinColumn(name = "empleado_id")
    private Empleado empleado;

    
}

File: icommerceApp/src/app/modelo/Fichaje.ts
import { Empleado } from "./Empleado";

export class Fichaje {
    constructor(public id: number,
        public horaEntrada: string,
        public activo: boolean,
        public horaSalida: string,
        public empleado: Empleado) { }

    public static createFromJsonObject(jsonObject: any): Fichaje {
        
        if(jsonObject['empleadoAsignado']!=null){
            console.log(jsonObject['empleadoAsignado']);
            
            let empleado:Empleado = Empleado.createFromJsonObject(jsonObject['empleadoAsignado']);
            console.log(empleado);
            
            let fichaje: Fichaje = new Fichaje(jsonObject['id'],
                jsonObject['horaEntrada'],
                jsonObject['activo'],
                jsonObject['horaSalida'],
                empleado);
            return fichaje;
        }else{
            let fichaje: Fichaje = new Fichaje(jsonObject['id'],
                jsonObject['horaEntrada'],
                jsonObject['activo'],
                jsonObject['horaSalida'],
                jsonObject['empleado']);
            return fichaje;
        }
       

    }
}